core/trino-main/src/main/java/io/trino/operator/join/unspilled/JoinProbe.java

```diff
@@ -108,7 +103,64 @@ public Page getPage()
         return page;
     }
 
-    private boolean currentRowContainsNull()
+    private long[] fillCache()
```

Comment:
this looks like a workaround for not batching `PageJoiner`. Do you plan to add batching there at some point?

Reply:
It is more complicated than it looks like. There is no 1:1 relation between probe and build side and the JoinProbe is the object responsible for handling that relation. So batching on any other level can be done only in certain conditions.
I do plan adding some quasi-batching capabilities to PageJoiner but do not consider this JoinProbe batching as a workaround.
I will post a draft PR tomorrow with some changes in PageJoiner so you can take a look.